tutor/src/components/annotations/hypothesis-store.js

```diff
@@ -0,0 +1,119 @@
+import HYPOTHESIS from '../../models/hypothesis';
+
+// Hypothesis token handshaking
+const hypothesisConfig = HYPOTHESIS.sidebarConfig().services[0];
+let hypothesisTokens;
+
+class HypothesisStore {
+
+  constructor() {
+    const gotToken = this.request({
+      mode: 'POST',
+      service: 'token',
+      headers: {
+        "Content-type": "application/x-www-form-urlencoded"
+      },
+      sendData: `grant_type=urn:ietf:params:oauth:grant-type:jwt-bearer&assertion=${hypothesisConfig.grantToken}`
+    }).then((response) => {
+      return this.accessToken = response.access_token;
+    });
+    const gotProfile = gotToken.then(() => {
+      return this.request({
+        mode: 'GET',
+        service: 'profile?authority=openstax.org'
+      }).then((response) => {
+        return this.userInfo = response.groups[0];
+      });
+    });
+
+    this.whenLoggedIn = Promise.all([gotToken, gotProfile]);
+  }
+
+  request(options) {
+    return new Promise((resolve, reject) => {
+      const xhr = new XMLHttpRequest();
+
+      xhr.open(options.mode, `${hypothesisConfig.apiUrl}${options.service}`, true);
+      if (this.accessToken) {
+        xhr.setRequestHeader('Authorization', `Bearer ${this.accessToken}`);
+      }
+      if (options.headers) {
+        for (const label of Object.keys(options.headers)) {
+          xhr.setRequestHeader(label, options.headers[label]);
+        }
+      }
+      xhr.onreadystatechange = function () {
+        if (this.readyState === 4 && this.status === 200) {
+          resolve(JSON.parse(this.response));
+        }
+        if (options.service === 'profile?authority=openstax.org' && this.readyState === 4 && this.status === 0) {
+          console.warn("Empty user info. Making something up.");
+          resolve({
+            groups: [{name: 'MadeUp', id:-999}]
+          });
+        }
+      };
+      xhr.send(options.sendData);
+    });
+  }
```

Comment:
we use https://github.com/axios/axios for xhr, and is already promise based.  That'd probably neaten this up.